packages/@aws-cdk/assert2/BUILD.md

```diff
@@ -0,0 +1,40 @@
+# How this package is built
+
+This package is built by copying the sources from `@aws-cdk/assert` and
+transforming the imports to depend on the v2 library, `aws-cdk-lib`, instead.
+
+Publishing of this package will only happen on the v2 branch. Just before
+publishing, we twiddle the `package.json` in the tarball and change the
+package name from `@aws-cdk/assert2` to `@aws-cdk/assert`, so that this
```

Comment:
🤢

Reply:
I wanted to throw another 'Why not just?' out there and be shot down.

Change the name of '@aws-cdk/assert' in master to '@aws-cdk/assert-internal' and make it private. Update all of our hypermodular packages to use this private package.
Then, have an '@aws-cdk/assert' clone and release as you've done here. 

The only difference between `master` and `v2-main` would then be the `package.json` and the dependencies it declares (in the former it'll be '@aws-cdk/core', etc. and latter 'aws-cdk-lib'). Maybe I'm oversimplifying?

Brings the complexity out onto the repo organization, but keeps the publishing/packing process standard. It feels like this name flip-flop around packing could introduce edge cases we don't foresee. (Nothing I can point at, just gut feeling)

Reply:
It's a good idea.

However, I am somewhat uncomfortable that in some cases we have source on `master` that seems to be v2-aware (like the init templates in `aws-cdk`, which have been duplicated for `v1` and `v2`) and in some cases we "just" make changes on `v2-main` in order to accomodate v2.

What's the rationale for these different treatments?

Reply:
Only in cases where something related to v2 is hard to capture on the master branch do we go directly `v2-main` branch. 
As far as I recall, these are the cases directly on v2-main branch: construct compat changes, [`release.json`](https://github.com/aws/aws-cdk/blob/v2-main/release.json) and [`FEATURE_FLAGS_EXPIRED`](https://github.com/aws/aws-cdk/blob/25b8eb48681919dbb0daa3a24623ca3911b6e358/packages/%40aws-cdk/cx-api/lib/features.ts#L138), (the last of which was an error on my part and can probably be mastered on 'master').
I could be missing something.

`package.json` differences are captured as `pkglint` rules and get autofixed as part of forward merge.

I'm curious - are you specifically uncomfortable that `package.json` between the branches will be different? Or is it something else?

Reply:
No, just a need to know what the heuristic is. Because right now I feel like a monkey in the dark.